miasm2/core/utils.py

```diff
@@ -73,7 +74,8 @@ def __init__(self, max_size, min_size=None, initialdata=None,
         self._min_size = min_size if min_size else max_size / 3
         self._max_size = max_size
         self._size = len(self._data)
-        self._counter = collections.Counter(self._data.keys())
+        # Do not use collections.Counter as it is quite slow
+        self._counter = dict((k, 1) for k in self._data.iterkeys())
```

Comment:
Also, you can iterate over the dictionary object directly, instead of calling `.iterkeys()`.